src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -73,11 +91,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     if (!request.getParameter("term").isEmpty()) {
       String term = request.getParameter("term");
-      Filter termFilter = new FilterPredicate("term", FilterOperator.EQUAL, term);
+      Filter termFilter;
+      if (fuzzy) {
+        List<String> terms = new ArrayList<>();
+        terms.add(getPrevTerm(term));
+        terms.add(getNextTerm(term));
+        termFilter = new FilterPredicate("term", FilterOperator.IN, terms);
+      } else {
+        termFilter = new FilterPredicate("term", FilterOperator.EQUAL, term);
+      }
       filters.add(termFilter);
     }
 
-    if (!request.getParameter("units").isEmpty()) {
+    if (!request.getParameter("units").isEmpty() && !(fuzzy)) {
```

Comment:
Shouldn't need () around fuzzy here I don't think unless Java is being extra weird